// ngx-malihu-scrollbar/fesm2015/ngx-malihu-scrollbar.js
import { __decorate } from 'tslib';
import { NgZone, Injectable, ElementRef, Input, Directive, NgModule } from '@angular/core';

let MalihuScrollbarService = class MalihuScrollbarService {
    constructor(zone) {
        this.zone = zone;
    }
    initScrollbar(element, options) {
        const jQueryElement = this.getElement(element);
        this.zone.runOutsideAngular(() => jQueryElement.mCustomScrollbar(options));
        if (jQueryElement.length > 0 && jQueryElement[0].tagName === 'BODY') {
            jQueryElement[0].style.position = 'absolute';
            jQueryElement[0].style.overflow = 'auto';
            jQueryElement[0].style.height = '100vh';
            jQueryElement[0].style.width = '100vw';
        }
    }
    scrollTo(element, parameter, options) {
        this.zone.runOutsideAngular(() => this.getElement(element).mCustomScrollbar('scrollTo', parameter, options));
    }
    update(element) {
        this.zone.runOutsideAngular(() => this.getElement(element).mCustomScrollbar('update'));
    }
    stop(element) {
        this.zone.runOutsideAngular(() => this.getElement(element).mCustomScrollbar('stop'));
    }
    disable(element, reset) {
        this.zone.runOutsideAngular(() => this.getElement(element).mCustomScrollbar('disable', !!reset));
    }
    destroy(element) {
        this.zone.runOutsideAngular(() => this.getElement(element).mCustomScrollbar('destroy'));
    }
    getElement(element) {
        if (typeof element === 'string') {
            return $(element);
        }
        if (typeof element === 'object' && element instanceof HTMLElement) {
            return $(element);
        }
        if (element instanceof jQuery || 'jquery' in Object(element)) {
            return element;
        }
        throw Error(`Unsupported element type in MalihuScrollbarService: ${element}`);
    }
};
MalihuScrollbarService.ctorParameters = () => [
    { type: NgZone }
];
MalihuScrollbarService = __decorate([
    Injectable()
], MalihuScrollbarService);

/**
 * Malihu Custom Scrollbar directive
 * Use this reference link for options definition
 * http://manos.malihu.gr/jquery-custom-content-scroller/
 */
let MalihuScrollbarDirective = class MalihuScrollbarDirective {
    constructor(elementRef, mScrollbarService) {
        this.elementRef = elementRef;
        this.mScrollbarService = mScrollbarService;
    }
    ngAfterViewInit() {
        this.initElements();
        this.initScrollbar();
    }
    ngOnDestroy() {
        this.destroyScrollbar();
    }
    initElements() {
        this.scrollableElement = !!this.scrollElementId
            ? $(`#${this.scrollElementId}`)
            : $(this.elementRef.nativeElement);
        if (!!this.scrollElementId && this.scrollableElement.length === 0) {
            console.error(`MalihuScrollbarDirective cannot find element with provided scrollElementId: ${this.scrollElementId}.`);
        }
    }
    initScrollbar() {
        this.mScrollbarService.initScrollbar(this.scrollableElement, this.scrollbarOptions);
    }
    destroyScrollbar() {
        try {
            this.mScrollbarService.destroy(this.scrollableElement);
        }
        catch (error) {
            // workaround for malihu-custom-scrollbar-plugin issue:
            // Cannot read property 'autoUpdate' of undefined
            // https://github.com/malihu/malihu-custom-scrollbar-plugin/issues/392
        }
    }
};
MalihuScrollbarDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: MalihuScrollbarService }
];
__decorate([
    Input()
], MalihuScrollbarDirective.prototype, "scrollElementId", void 0);
__decorate([
    Input()
], MalihuScrollbarDirective.prototype, "scrollbarOptions", void 0);
MalihuScrollbarDirective = __decorate([
    Directive({
        selector: '[MalihuScrollbar], [malihu-scrollbar]',
    })
], MalihuScrollbarDirective);

var MalihuScrollbarModule_1;
let MalihuScrollbarModule = MalihuScrollbarModule_1 = class MalihuScrollbarModule {
    static forRoot() {
        return {
            ngModule: MalihuScrollbarModule_1,
            providers: [MalihuScrollbarService],
        };
    }
    static forChild() {
        return {
            ngModule: MalihuScrollbarModule_1,
        };
    }
};
MalihuScrollbarModule = MalihuScrollbarModule_1 = __decorate([
    NgModule({
        exports: [MalihuScrollbarDirective],
        declarations: [MalihuScrollbarDirective],
    })
], MalihuScrollbarModule);

/**
 * Generated bundle index. Do not edit.
 */

export { MalihuScrollbarDirective, MalihuScrollbarModule, MalihuScrollbarService };
